extension/src/prototypes/view/environments/MapEnvironment.tsx

```diff
@@ -0,0 +1,57 @@
+// import { VectorMapImageryLayer } from "@takram/plateau-datasets";
+import { useAtomValue } from "jotai";
+import { type FC } from "react";
+
+import { Scene, SceneProps } from "../../../shared/reearth/scene";
+import { type ColorMode } from "../../shared-states";
+import { enableTerrainLightingAtom } from "../states/app";
+
+// Flat white sky and gray ground
+const sphericalHarmonicCoefficients: [x: number, y: number, z: number][] = [
+  [0.651181936264038, 0.651181936264038, 0.651181936264038], // L00, irradiance, pre-scaled base
+  [0.335859775543213, 0.335859775543213, 0.335859775543213], // L1-1, irradiance, pre-scaled base
+  [0.000000874592729, 0.000000874592729, 0.000000874592729], // L10, irradiance, pre-scaled base
+  [0.000000027729817, 0.000000027729817, 0.000000027729817], // L11, irradiance, pre-scaled base
+  [0.000000014838997, 0.000000014838997, 0.000000014838997], // L2-2, irradiance, pre-scaled base
+  [-0.000000005038311, -0.000000005038311, -0.000000005038311], // L2-1, irradiance, pre-scaled base
+  [0.000121221753943, 0.000121221753943, 0.000121221753943], // L20, irradiance, pre-scaled base
+  [0.000000282587223, 0.000000282587223, 0.000000282587223], // L21, irradiance, pre-scaled base
+  [0.000364663166692, 0.000364663166692, 0.000364663166692], // L22, irradiance, pre-scaled base
+];
+
+export interface MapEnvironmentProps extends SceneProps {
+  colorMode?: ColorMode;
+}
+
+export const MapEnvironment: FC<MapEnvironmentProps> = ({ colorMode = "light", ...props }) => {
+  // invariant(
+  //   process.env.NEXT_PUBLIC_TILES_BASE_URL != null,
+  //   "Missing environment variable: NEXT_PUBLIC_TILES_BASE_URL",
+  // );
+  const enableTerrainLighting = useAtomValue(enableTerrainLightingAtom);
+
+  // const [layer, setLayer] = useState<ImageryLayerHandle | null>(null);
+  // useEffect(() => {
+  //   layer?.sendToBack();
+  // }, [layer]);
+
+  return (
+    <Scene
+      // TODO: Define in theme
+      // TODO: Swap background when view is ready
+      globeBaseColor={colorMode === "light" ? "#f7f7f7" : "#000000"}
+      enableGlobeLighting={enableTerrainLighting}
+      lightIntensity={10}
+      shadowDarkness={colorMode === "light" ? 0.7 : 0.3}
+      imageBasedLightingIntensity={1}
+      sphericalHarmonicCoefficients={sphericalHarmonicCoefficients}
+      showSkyBox={false}
+      atmosphereSaturationShift={-1}
+      groundAtmosphereBrightnessShift={2}
+      // TODO(ReEarth): Use Takram's tile
+      // TODO(ReEarth): Support tile brightness
+      tiles={[{ id: "default", tile_type: "stamen_toner" }]}
```

Comment:
I think the array need to be wrapped with useMemo, it's causing update on each frame now.